cdap-docs/developers-manual/source/building-blocks/workers.rst

```diff
@@ -82,7 +82,48 @@ application in the same namespace. WorkerContext can be used to discover the URL
   @Override
   public void run() {
     URL url = getContext().getServiceURL("myService");
-    //To discover a Service in another application in the same namespace, use:
+
+    // To discover a Service in another application in the same namespace, use:
     url = getContext().getServiceURL("anotherAppName", "anotherServiceId");
   }
 
+Writing to Streams (Beta)
+=========================
+
+Workers have the ability to write to ``Streams`` through the ``WorkerContext``. The implementation internally
+issues an HTTP request to the Stream Service to persist the data. Because of that, a write to a Stream
+cannot be rolled back, and thus the write differs in semantics compared to writing to ``Datasets`` from inside the
+``run`` method of a ``TxRunnable``.
+
+The write operation throws an ``IOException`` if it could not write to a ``Stream``. Writing to streams can be
+performed as either single event writes or in batch.
+
+When uploading events in batch, there are two options: either uploading a ``File`` or writing multiple events
+through a ``StreamBatchWriter``. In batch mode, the content type of the data must be specified. Refer
+to the :ref:`Stream RESTful API <http-restful-api-stream>` for information on the content type specification.
+
+With a ``StreamBatchWriter``, the ``close`` method` needs to be called after all the writes have been performed::
+
+  @Override
+  public void run() {
+    try {
+      // Writing a single string event to stream myStream
+      getContext().write("myStream", "data1");
+
+      Map<String, String> header = Maps.newHashMap();
+      header.put("k1", "v1");
+
+      // Writing a single string event with header to stream myStream
+      getContext().write("myStream", "data2", header);
+
+      // Writing a set of events as one batch operation to stream myStream (with content type as text)
+      StreamBatchWriter batchWriter = getContext().createBatchWriter("myStream", "text/string");
```

Comment:
As we discussed, adding @Beta annotations to all methods is not the standard practice. I will add some comment to the class' Javadoc. I am concerned since this is not a common practice. Let me know if still think it would be useful.